src/pathology-report/ReportComponent.tsx

```diff
@@ -1,40 +1,476 @@
+import React from "react";
+import cloneDeep from "lodash-es/cloneDeep";
+import {
+  DataTable,
+  Table,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableCell,
+} from "carbon-components-react";
+
+import styles from "./ReportComponent.css";
+// import { TableRow } from './TableRow';
+import {
+  getUserLocation,
+  getConceptAnswers,
+  getLocations,
+  getEncounters,
+  postSampleDropoffObs,
+  voidSampleDropoff,
+  postSampleStatusChangeObs,
+  updateSampleStatusChangeObs,
+  postReferralStatusChangeObs,
+  updateReferralStatusChangeObs,
+} from "./ReportComponent.resource";
+import {
+  RWINKWAVUHEALTHCENTERNAME,
+  SAMPLESTATUSUUID,
+  REFERRALSTATUSUUID,
+  SAMPLEDROPOFFUUID,
+  YESCONCEPTNAME,
+} from "../constants.js";
 
 const ReportComponent = () => {
+  const [userLocation, setUserLocation] = React.useState();
   const [encountersList, setEncountersList] = React.useState([]);
-  // const [selectedEncounter, setSelectedEncounter] = React.useState();
+  const tempEncList = React.useRef([]);
+  const [sendingHospital, setSendingHospital] = React.useState("");
+  const [listOfHospitals, setListOfHospitals] = React.useState([]);
+  const [sampleStatusResults, setSampleStatusResults] = React.useState([]);
+  const [referralStatusResults, setReferralStatusResults] = React.useState([]);
+  const [sampleStatus, setSampleStatus] = React.useState("");
+  const [referralStatus, setReferralStatus] = React.useState("");
+  const [patientName, setpatientName] = React.useState("");
+  // var rows =[];
+  const headers = [
+    // {
+    //   key: 'patientLink',
+    //   header: 'Link to patient',
+    // },
+    {
+      key: "patientNames",
+      header: "Patient name",
+    },
+    {
+      key: "pathologyRequest",
+      header: "pathology request",
+    },
+    {
+      key: "SendingHospital",
+      header: "Sending Hospital",
+    },
+    {
+      key: "phoneNumber",
+      header: "Phone number",
+    },
+    {
+      key: "sampleStatus",
+      header: "Sample status",
+    },
+    {
+      key: "dateOfRequest",
+      header: "Date of Request",
+    },
+    {
+      key: "referralStatus",
+      header: "Referral status",
+    },
+    {
+      key: "sampleDropOff",
+      header: "Sample drop off?",
+    },
+  ];
 
   React.useEffect(() => {
-    getPatientEncounters('').then(setEncountersList);
+    getUserLocation().then(setUserLocation);
+    getLocations().then(setListOfHospitals);
+    getConceptAnswers(SAMPLESTATUSUUID).then(setSampleStatusResults);
+    getConceptAnswers(REFERRALSTATUSUUID).then(setReferralStatusResults);
+    getEncounters().then(setEncountersList);
   }, []);
 
+  const handleSampleStatus = (value) => {
+    setSampleStatus(value);
+  };
+  const handleReferralStatus = (value) => {
+    setReferralStatus(value);
+  };
+  const handleSendingHospital = (value) => {
+    setSendingHospital(value);
+  };
+  const handlePatientName = (value) => {
+    setpatientName(value);
+  };
```

Comment:
```suggestion
```

Since `handleSampleStatus` is exactly `setSampleStatus`, you can just use `setSampleStatus`.

Reply:
Sure.